src/fromager/commands/bootstrap.py

```diff
@@ -77,19 +87,41 @@ def bootstrap(
         )
     logger.info("bootstrapping %r variant of %s", wkctx.variant, to_build)
 
+    prev_graph = None
+    if prev_graph_file:
+        prev_graph = dependency_graph.DependencyGraph.from_file(prev_graph_file)
+
     pre_built = wkctx.settings.list_pre_built()
     if pre_built:
         logger.info("treating %s as pre-built wheels", sorted(pre_built))
 
     server.start_wheel_server(wkctx)
 
+    # we need to resolve all the top level dependencies before we start bootstrapping.
+    # this is to ensure that if we are using an older bootstrap to resolve packages
+    # we are able to upgrade a package anywhere in the dependency tree if it is mentioned
+    # in the toplevel without having to fall back to history
+    for req in to_build:
```

Comment:
What happens if a prebuilt  appears in this list?

Reply:
ah nice catch this is why the e2e test was failing. Thank you!